python/lsst/ts/dream/csc/dream_csc.py

```diff
@@ -132,43 +178,129 @@ async def begin_disable(self, id_data: salobj.BaseDdsDataType) -> None:
 
     async def do_resume(self, data: salobj.BaseMsgType) -> None:
-        # To be implemented later.
-        pass
+        if not self.connected:
+            await self.connect()
+
+    async def weather_loop(self) -> None:
+        """Periodically checks weather station, and sends a flag if needed.
+
+        A weather flag is sent when the weather has changed, or every ten
+        minutes.
+        """
+        while self.model is not None and self.model.connected:
+            if self.simulation_mode == 0:
+                await self.connect_ess_remote()  # Connect to ESS if not already connected
+                if self.ess_remote is None:
+                    self.log.error("Failed to connect to weather CSC.")
+                    continue
+
+                try:
+                    # Get weather data.
+                    weather_ok_flag = True
+                    air_flow = await self.ess_remote.tel_airFlow.next(
+                        flush=False, timeout=2
+                    )
+                    if air_flow is None or air_flow.speed > 25:
+                        weather_ok_flag = False
+
+                    precipitation = await self.ess_remote.evt_precipitation.get()
+                    if precipitation is None or (
+                        precipitation.raining or precipitation.snowing
+                    ):
+                        weather_ok_flag = False
+                except Exception:
+                    self.log.exception("Failed to read weather data from ESS.")
+                    self.ess_remote = None
+                    continue
+
+            else:
+                # In simulation mode, don't try to read the weather station.
+                weather_ok_flag = True
+
+            # Compare weather flag with cached value.
+            time_since_last_update = time.time() - self.last_weather_update
+            if (
+                weather_ok_flag != self.weather_ok_flag
+                or time_since_last_update > 60 * 10
+            ):
+                try:
+                    # Send weather flag
+                    if self.model is not None:
+                        await self.model.set_weather_ok(weather_ok_flag)
+                        self.last_weather_update = time.time()
+                        self.weather_ok_flag = weather_ok_flag
+                    else:
+                        self.log.info(
+                            "Weather loop ending because of TCP disconnection."
+                        )
+                        return
+                except asyncio.CancelledError:
+                    self.log.info("Weather loop ending because of asycio cancel.")
+                    return
+                except Exception:
+                    self.log.exception("Failed to send weather flag!")
+
+            # Sleep for a bit.
+            try:
+                self.weather_sleep_task = asyncio.ensure_future(
+                    asyncio.sleep(10.0)  # TODO: Make this configurable.
```

Comment:
All TODOs require a DM ticket. Please open one and mention the ticket here so this will not be forgotten.

Reply:
Here's the ticket: https://rubinobs.atlassian.net/issues/DM-47503?filter=-2

Reply:
With the modification of the settings for ess_index I decided to just throw this in there as well.